downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/GenerateGOAnnotationFile.java

```diff
@@ -0,0 +1,413 @@
+                                }
+                                processProteins(goTerm, proteinInstances, reactionInst, catalystInst);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        outputGOAFile();
+        Files.move(Paths.get(GOA_FILENAME), Paths.get(releaseNumber + "/" + GOA_FILENAME), StandardCopyOption.REPLACE_EXISTING);
+        logger.info("Finished generating gene_association.reactome");
+    }
+
+    // This method uses an AttributeQueryRequest to find all Proteins (above and below in the hierarchy) associated with a ReactionlikeEvent.
+    private static Set<GKInstance> getReactionProteins(GKInstance reactionInst) throws Exception {
+        List<ClassAttributeFollowingInstruction> classesToFollow = new ArrayList<>();
+        classesToFollow.add(new ClassAttributeFollowingInstruction(ReactomeJavaConstants.Pathway, new String[]{ReactomeJavaConstants.hasEvent}, new String[]{}));
+        classesToFollow.add(new ClassAttributeFollowingInstruction(ReactomeJavaConstants.ReactionlikeEvent, new String[]{ReactomeJavaConstants.input, ReactomeJavaConstants.output, ReactomeJavaConstants.catalystActivity}, new String[]{}));
+        classesToFollow.add(new ClassAttributeFollowingInstruction(ReactomeJavaConstants.Reaction, new String[]{ReactomeJavaConstants.input, ReactomeJavaConstants.output, ReactomeJavaConstants.catalystActivity}, new String[]{}));
+        classesToFollow.add(new ClassAttributeFollowingInstruction(ReactomeJavaConstants.CatalystActivity, new String[]{ReactomeJavaConstants.physicalEntity}, new String[]{}));
+        classesToFollow.add(new ClassAttributeFollowingInstruction(ReactomeJavaConstants.Complex, new String[]{ReactomeJavaConstants.hasComponent}, new String[]{}));
+        classesToFollow.add(new ClassAttributeFollowingInstruction(ReactomeJavaConstants.EntitySet, new String[]{ReactomeJavaConstants.hasMember}, new String[]{}));
+        classesToFollow.add(new ClassAttributeFollowingInstruction(ReactomeJavaConstants.Polymer, new String[]{ReactomeJavaConstants.repeatedUnit}, new String[]{}));
+
+        // All EWAS' associated with each of the above-stated classes will be output for this ReactionlikeEvent.
+        String[] outClasses = new String[]{ReactomeJavaConstants.EntityWithAccessionedSequence};
+        return InstanceUtilities.followInstanceAttributes(reactionInst, classesToFollow, outClasses);
+    }
+
+    // Retrieves all protein instances from an EntitySet comprised of only EWAS' or that are not a Complex/Polymer but are an EWAS.
+    private static Set<GKInstance> getMolecularFunctionProteins(GKInstance entityInst) throws Exception {
+        Set<GKInstance> proteinInstances = new HashSet<>();
+        SchemaClass entitySchemaClass = entityInst.getSchemClass();
+        if (!(entitySchemaClass.isa(ReactomeJavaConstants.Complex) || entitySchemaClass.isa(ReactomeJavaConstants.Polymer))) {
```

Comment:
I wonder if this check for complex or polymer is needed since the inner conditions look to see if the entity schema class is a set or EWAS?  No action should be taken if it is a complex or polymer even if the check is removed.